controllers/quote.js

```diff
@@ -121,16 +125,29 @@ const UnacceptableExpiryError = require('../errors/unacceptable-expiry-error')
 /* eslint-enable */
 
 exports.get = function *() {
-  let quote = yield backend.getQuote({
-    source_ledger: this.query.source_ledger,
-    destination_ledger: this.query.destination_ledger,
-    source_amount: this.query.source_amount,
-    destination_amount: this.query.destination_amount
-  })
+  const query = new QuoteQuery(this.query)
+  yield query.loadLedgers()
+  const quote = yield backend.getQuote(query.toQuoteArgs())
+
+  const sourceBalance = yield getConnectorBalance(query.source_ledger)
+  const sourceAmount = new BigNumber(quote.source_amount)
+  if (sourceBalance.lessThan(sourceAmount)) {
+    throw new UnacceptableAmountError('Insufficient liquidity in market maker account')
+  }
+
+  log.debug('' +
+    quote.source_amount.toFixed(2) + ' ' +
+    query.source_ledger + ' => ' +
+    quote.destination_amount.toFixed(2) + ' ' +
+    query.destination_ledger)
+
+  this.body = query.toPaymentTemplate(quote)
+}
 
+function QuoteQuery (params) {
```

Comment:
What I mean is that, it doesn't seem to belong as part of the interface of a `QuoteQuery`. It would be more natural to have a function that fetches ledgers, that receives a `QuoteQuery` as an argument.